### ISSUE TITLE
Switch RELEASE.rst system to use Towncrier

### ISSUE THREAD

**Issue by DRMacIver** [HypothesisWorks/hypothesis-python#929]
Apparently there is a relatively widespread project for doing what I NIHed with our RELEASE.rst management: https://github.com/hawkowl/towncrier

We should think about using this ourselves. It seems to be a strict superset of our current functionality, and it would be nice to be on a more standard system rather than an ad hoc hack.

**Comment by DRMacIver** [comment]
(We'll have to think about how to rebuild the RELEASE_TYPE functionality on top of this if we do this)

**Comment by Zac-HD** [comment]
I've played with towncrier before, and don't actually like it much - it's designed to make maintaining an upcoming changelog easier with many people contributing patches/features/bugfixes/docs/etc which will all go in one release.  AFAICT it wouldn't fit our one-PR-equals-one-release pattern well at all; both because we can't take advantage of it's strengths and because it's a lot more hassle to get the build and deploy right.

TLDR; I like our `RELEASE.rst` system just fine.

**Comment by alexwlchan** [comment]
> it would be nice to be on a more standard system rather than an ad hoc hack

An ad hoc hack which, by the way, has now worked pretty well for several dozen releases. 😉

**Comment by DRMacIver** [comment]
> An ad hoc hack which, by the way, has now worked pretty well for several dozen releases. 

Well, sure. That's the Hypothesis style of ad hoc hac. ;-)

**Comment by alexwlchan** [comment]
Subtext to my previous post:

This ad hoc hac has had a sufficiently good track record that I’d be wary about fiddling with it again.

We'll likely break something, and it sounds like towncrier isn’t a drop-in replacement – we’d be just be shuffling around the complexity.

**Comment by DRMacIver** [comment]
Fair. I shall consider myself outvoted. :-)
